google-cloud-logging/src/main/java/com/google/cloud/logging/LoggingHandler.java

```diff
@@ -372,34 +401,36 @@ public void setFlushLevel(Level flushLevel) {
    * @param synchronicity {@link Synchronicity}
    */
   public void setSynchronicity(Synchronicity synchronicity) {
-    getLogging().setWriteSynchronicity(synchronicity);
+    logging.setWriteSynchronicity(synchronicity);
   }
 
   /** Get the flush log level. */
   public Synchronicity getSynchronicity() {
-    return getLogging().getWriteSynchronicity();
+    return logging.getWriteSynchronicity();
   }
 
   /** Sets the metadata auto population flag. */
-  public void setAutoPopulateMetadata(Boolean value) {
-    checkNotNull(value);
+  public void setAutoPopulateMetadata(boolean value) {
     this.autoPopulateMetadata = value;
-    List<WriteOption> writeOptions = Arrays.asList(defaultWriteOptions);
-    for (int i = 0; i < writeOptions.size(); i++) {
-      if (writeOptions.get(i).getOptionType() == WriteOption.OptionType.AUTO_POPULATE_METADATA) {
-        writeOptions.remove(i);
-        break;
-      }
-    }
-    writeOptions.add(WriteOption.autoPopulateMetadata(value));
-    defaultWriteOptions = Iterables.toArray(writeOptions, WriteOption.class);
   }
 
   /** Gets the metadata auto population flag. */
   public Boolean getAutoPopulateMetadata() {
     return this.autoPopulateMetadata;
   }
 
+  /**
+   * Enable/disable use of structured logging. When enabled, logs will be printed to STDOUT in the
+   * structured logging format. Otherwise, logs will be ingested using Logging API.
+   */
+  public void setUseStructuredLogging(boolean value) {
```

Comment:
Reading the comment, `setUseStructuredLogging` name is misleading - in both cases we use structured logging but when `setUseStructuredLogging` is true we just printing to STDOUT. I would change it to something like `setPrintStructuredLoggingToStdout` or so

Reply:
Good catch. Indeed, our internal use of the term does not exactly correlate with what public documentation says:
> structured logs refer to log entries that use the jsonPayload field to add structure to their payloads.

The flag name will be changed to `useLoggingAgent`. WDYT?